airflow/models.py

```diff
@@ -2592,6 +2594,18 @@ def xcom_pull(
             dag_id=dag_id,
             include_prior_dates=include_prior_dates)
 
+    def get_extra_links(self, dttm, redirect_to):
+        """
+        For an operator, gets the URL that the external links specified in
+        `extra_links` should point to.
+        :raise ValueError: The error message of a ValueError will be passed on through to
+            the fronted to show up as a tooltip on the disabled link
+        :param dttm: The datetime parsed execution date for the URL being searched for
+        :param redirect_to: The name of the link we're looking for the URL for. Should be
+            one of the options specified in `extra_links`
+        :return: A URL
```

Comment:
Haven't dug deep in the PR yet, but how's the button label defined? Maybe this function could return a tuple of (button_label, url, button_tooltip)

Reply:
The button label is defined at the top of the operator, similar to `template_fields`. Because we're using asynchronous responses from the frontend, we need to define the link names statically so that they show up as a button in the UI

Reply:
The button label is defined at the top of the operator, similar to `template_fields`. Because we're using asynchronous responses from the frontend, we need to define the link names statically so that they show up as a button in the UI

Reply:
The button label is defined at the top of the operator, similar to `template_fields`. Because we're using asynchronous responses from the frontend, we need to define the link names statically so that they show up as a button in the UI